aten/src/ATen/native/cuda/Blas.cpp

```diff
@@ -277,7 +265,8 @@ Tensor& addmm_out_cuda_impl(Tensor& result, const Tensor& self, const Tensor& ma
         isSupportedHipLtROCmArch(self.device().index()) &&
         (scalar_type == at::ScalarType::Float ||
           scalar_type == at::ScalarType::Half ||
-          scalar_type == at::ScalarType::BFloat16);
+          scalar_type == at::ScalarType::BFloat16) &&
+        beta.toComplexDouble() == 0.0;
```

Comment:
@jeffdaily I wonder if C and D are the same pointer this would work. To deal with mm, I'm just checking if beta==0; for inplace addmm, beta != 0 so it falls back to hipblas and take a perf hit.